api/register.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mhghw/fara-message/jwt"
+)
+
+type RegUser struct {
+	ID              string `json:"id"`
+	Username        string `json:"username"`
+	FirstName       string `json:"first_name"`
+	LastName        string `json:"last_name"`
+	Password        string `json:"password"`
+	ConfirmPassword string `json:"confirm_password"`
+	Gender          string `json:"gender"`
+	DateOfBirth     string `json:"date_of_birth"`
+}
+
+// handle func for register,receives a json string according to the RegUser struct and generate a JWT with ID parameter
+func Register(c *gin.Context) {
+	var requestBody RegUser
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+
+		log.Print(http.StatusBadRequest, err)
+	}
+	if requestBody.Password == requestBody.ConfirmPassword {
+		token, err := jwt.CreateToken(requestBody.ID)
+		if err != nil {
+			panic("failed to create token")
+		}
+		c.JSON(http.StatusOK, token)
+		fmt.Println(requestBody)
+	} else {
+		c.JSON(http.StatusNotAcceptable, "password does not match")
```

Comment:
You are sending a string in c.JSON that returns a JSON response to the client. I don't want to tell you which errors you should return to client, but if you are returning a string, do it correctly.
Also each http status has a definition. I don't think [StatusNotAcceptable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) is a proper status code here.
Usually we return bad request status code, if there was some input validation issues.